docs/mca/set-mca-param.rst

```diff
@@ -0,0 +1,251 @@
+
+The contents of tuned files consist of one or more lines, each of
+which contain zero or more `-x` and `--mca` options.  Comments are not
+allowed.  For example, the following tuned file:
+
+.. code-block::
+
+   -x envvar1=value1 -mca param1 value1 -x envvar2
+   -mca param2 value2
+   -x envvar3
+
+is equivalent to:
+
+.. code-block:: sh
+
+   shell$ mpirun \
+       -x envvar1=value1 -mca param1 value1 -x envvar2 \
+       -mca param2 value2
+       -x envvar3 \
+       ...rest of mpirun command line...
+
+Although the typical use case for tuned parameter files is to be
+specified on the command line, they can also be set as MCA parameters
+in the environment.  The MCA parameter ``mca_base_envvar_file_prefix``
+contains a comma-delimited list of tuned parameter files exactly as
+they would be passed to the ``--tune`` command line option.  The MCA
+parameter ``mca_base_envvar_file_path`` specifies the path to search
+for tuned files with relative paths.
+
+.. error:: TODO Check that these MCA var names ^^ are correct.
+
+.. _label-mca-set-param-config-file:
+
+Configuration files
+-------------------
+
+Finally, simple configuration text files can be used to set MCA
+parameter values.  Parameters are set one per line (comments are
+permitted).  For example:
+
+.. code-block:: ini
+
+   # This is a comment
+   # Set the same MCA parameter as in previous examples
+   mpi_show_handle_leaks = 1
+
+Note that quotes are *not* necessary for setting multi-word values
+in MCA parameter files.  Indeed, if you use quotes in the MCA
+parameter file, they will be used as part of the value itself.  For
+example:
+
+.. code-block:: ini
+
+   # The following two values are different:
+   param1 = value with multiple words
+   param2 = "value with multiple words"
+
+By default, two files are searched (in order):
+
+#. ``$HOME/.openmpi/mca-params.conf``: The user-supplied set of
+   values takes the highest precedence.
+#. ``$prefix/etc/openmpi-mca-params.conf``: The system-supplied set
+   of values has a lower precedence.
+
+More specifically, the MCA parameter ``mca_param_files`` specifies a
+colon-delimited path of files to search for MCA parameters.  Files to
+the left have lower precedence; files to the right are higher
+precedence.
+
+.. note:: Keep in mind that, just like components, these parameter
+          files are *only* relevant where they are "visible"
+          (:ref:`see this FAQ entry
+          <faq-general-tuning-install-components>`).  Specifically,
+          Open MPI does not read all the values from these files
+          during startup and then send them to all nodes in the job.
+          Instead, the files are read on each node during each
+          process' startup.
+
+          *This is intended behavior:* it allows for per-node
+          customization, which is especially relevant in heterogeneous
+          environments.
+
+.. error:: TODO This table needs to be checked for correctness.
```

Comment:
Can you check this?

Reply:
Will do

Reply:
.prrte needs to be .prte
and the file for open mpix is - pmix-mca-params.conf not openpmix-mca-params.conf